File: Js/MergeSort.js
const MergeSort = (l) => {
  if (l.length < 2) {
    return l;
  }
  const mid = Math.floor(l.length / 2);
  const left = l.slice(0, mid);
  const right = l.slice(mid);
  return Merge(MergeSort(left), MergeSort(right));
};
const Merge = (left, right) => {
  let result = [];
  let leftPointer = 0;
  let rightPointer = 0;
  while (leftPointer < left.length && rightPointer < right.length) {
    if (left[leftPointer] > right[rightPointer]) {
      result.push(right[rightPointer]);
      rightPointer++;
    } else {
      result.push(left[leftPointer]);
      leftPointer++;
    }
  }
  if (leftPointer === left.length) {
    result.push(...right.slice(rightPointer));
  } else {
    result.push(...left.slice(leftPointer));
  }
  return result;
};

const l = [2, 5, 1, 7, 3, 9, 43, 12, 13, 13, 54, 23, 11];
console.log(MergeSort(l));

File: Python/MergeSort.py
#MergeSort - Top3
def MergeSort(l):
    #Key recursive
    if len(l)==1:
        return l
    
    #Divide list on left and right side
    mid=len(l)//2
    left,right=l[:mid],l[mid:]
    
    #Do recursive
    return Merge(MergeSort(left),MergeSort(right))
    
def Merge(left,right):
    leftPointer,rightPointer=0,0
    result=[]
    
    #Do cycle until one of list give all elements
    while len(left)>leftPointer and len(right)>rightPointer:
        if left[leftPointer]>right[rightPointer]:
            result.append(right[rightPointer])
            rightPointer+=1
        else:
            result.append(left[leftPointer])
            leftPointer+=1
    
    #extend another elements
    if leftPointer==len(left):result.extend(right[rightPointer:])
    else:result.extend(left[leftPointer:])
    return result
    
#Create ourList and Check MergeSort
ourList=[2,6,7,12,64,23]
print(MergeSort(ourList))
    